src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -1952,24 +1990,37 @@ static int addr_exchange_root_vni(MPIR_Comm * init_comm)
     int mpi_errno = MPI_SUCCESS;
     int size = MPIR_Process.size;
     int rank = MPIR_Process.rank;
+    int num_nics = MPIDI_OFI_global.num_nics;
```

Comment:
Either consistently use `num_nics` or consistently use `MPIDI_OFI_global.num_nics`, but not both.

Reply:
Agreed. Not sure why this is such a mess...